### PR TITLE
show valid reports

### PR DESCRIPTION
**Pull request by flamingbear**
**Summary:** Updates single reconciliation report view to follow design.  Including a badge for Success/Conflict for the entire report. Adds collapsable cards for each of the different comparisons for Cumulus's holdings vs CMR.

Addresses [CUMULUS-1905:show valid reports](https://bugs.earthdata.nasa.gov/browse/CUMULUS-1905)

## Changes

* Adds report-reshaper function to pullout and configure the returned reconciliation report for the display.
* refactors num--status => num-status (sorry)
* refactors status_badge__{modifier} => status__badge--{modifier}
* updates reconciliationReport to display reports in an intuitive manner that matches the design.

## PR Checklist

- [x] Update CHANGELOG
- [x] Unit tests
- [x] Adhoc testing
- [x] Integration tests

### CODE REVIEW

**Comment by jjmccoy** [comment]
UI/UX: Looks good and I know this wasn't easy. I just have a few observations below.

PI 20 2-Cumulus Dashboard - Individual Report - Inventory - Cumulus Tab

- Missing the 'Search' input for easier discovery and this is a consistent element for all tables

- There are some headings and body text that are missing that give context and for flow of the page

- The 'Expand/Collapse' button needs to be right justified because it goes inline with where the the buttons of the accordion are which are on the right 

There are some styling tweaks with padding and spacing but can be handled later.

**Comment by flamingbear** [comment]
@jjmccoy 

The AC for CUMULUS-1905 is: "The overall status properly calculates from the bucket status to show on the UI."


"Missing the 'Search' input for easier discovery and this is a consistent element for all tables"

There is no "search" or "table filters" functionality on the tables that existed.  This ticket just wrapped those.  I assumed that work was to be done in another ticket, because it seems outside the scope of this one.  If there's not a ticket for it, I'll write one.  There is a search and filter on the overview page, that allows for narrowing down the desired set of reconciliation reports before selecting one. 

As for "headings and body text that are missing that give context and for flow of the page"  Not sure which headings of text you're speaking of.  Can you be clearer on that?

**Comment by jjmccoy** [comment]
> @jjmccoy
> 
> The AC for CUMULUS-1905 is: "The overall status properly calculates from the bucket status to show on the UI."
> 
> "Missing the 'Search' input for easier discovery and this is a consistent element for all tables"
> 
> There is no "search" or "table filters" functionality on the tables that existed. This ticket just wrapped those. I assumed that work was to be done in another ticket, because it seems outside the scope of this one. If there's not a ticket for it, I'll write one. There is a search and filter on the overview page, that allows for narrowing down the desired set of reconciliation reports before selecting one.
> 
Okay. The search and filter are needed for the items in the table. I would suggest that can you please put in a ticket for the search and filter module in a report view to be built.

> As for "headings and body text that are missing that give context and for flow of the page" Not sure which headings of text you're speaking of. Can you be clearer on that?
I guess we can reserve this for the catch all style ticket (CUMULUS-1907).

**Comment by flamingbear** [comment]
> Okay. The search and filter are needed for the items in the table. I would suggest that can you please put in a ticket for the search and filter module in a report view to be built.

I can put in that ticket and ask Lauren to prioritize.

> > As for "headings and body text that are missing that give context and for flow of the page" Not sure which headings of text you're speaking of. Can you be clearer on that?
> 
> I guess we can reserve this for the catch all style ticket (CUMULUS-1907).

Is it something that is obvious that I missed, because I don't mind fixing it. Or I can include it in the new ticket as well.

**Comment by jjmccoy** [comment]
UI/UX has reviewed the UI functionality and layout. LGTG. Need a logic review.

File: app/src/js/components/ReconciliationReports/reconciliation-report.js [hunk 1]
```diff
@@ -1,269 +1,264 @@
...
+            activeCard={activeIdx}
+          />
+        </div>
+      </section>
+
+      <section className="page__section">
+        <div className="multicard">
+          <div className="collapse-link">
+            <span className="link" onClick={handleExpandClick}>
+              {!allCollapsed() ? 'Expand All' : 'Collapse All'}
```
Comment (dopeters) [inline app/src/js/components/ReconciliationReports/reconciliation-report.js:195]: nice

File: app/src/js/components/ReconciliationReports/reconciliation-report.js [hunk 2]
```diff
@@ -1,269 +1,264 @@
...
+    active: true,
+  },
+];
+
+const ReportStateHeader = ({ reportState, startDate, endDate }) => {
+  const displayStartDate = startDate
+    ? new Date(startDate).toLocaleDateString()
+    : 'missing';
+  const displayEndDate = endDate
+    ? new Date(startDate).toLocaleDateString()
```
[the remark this answers is not in the document]
Comment (flamingbear) [reply app/src/js/components/ReconciliationReports/reconciliation-report.js]: Yup.  Oops.

**Pull request merged**